src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -447,7 +447,14 @@ public void publish(Stream stream, Map<String, Object> payload, Date timestamp,
 
     public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String partitionKey, UnencryptedGroupKey newGroupKey) {
```

Comment:
Should we simply add `synchronized` keyword to `publish`? That would secure the rest of the method from race conditions (access to `keyExchangeUtil` on lines 463-465 for example).

Reply:
As discussed in Slack. let's leave this out for now and re-consider later.